src/app/shared/case-editor/page-validation.service.ts

```diff
@@ -0,0 +1,41 @@
+import { Injectable } from '@angular/core';
+import { CaseField } from '../domain/definition/case-field.model';
+import { AbstractControl, FormGroup } from '@angular/forms';
+import { ShowCondition } from '../conditional-show/conditional-show.model';
+import { CaseFieldService } from '../domain/case-field.service';
+import { WizardPage } from '../domain/wizard-page.model';
+
+@Injectable()
+export class PageValidationService {
+  constructor(private caseFieldService: CaseFieldService) {}
+
+  isPageValid(page: WizardPage, editForm: FormGroup): boolean {
+    return page.case_fields
+      .filter(caseField => !this.caseFieldService.isReadOnly(caseField))
```

Comment:
Fatih can you please add a TODO comment inside CaseFieldService to get rid of CaseFieldService and move all its methods inside CaseField itself please?